pixelmatch.py

```diff
@@ -29,7 +29,7 @@ def pixelmatch(img1, img2, width: int, height: int, output=None, options=None):
     image_len = width * height
     identical = True
 
-    for i in range(image_len):
+    for i in range(len(img1)):
```

Comment:
https://github.com/whtsky/pixelmatch-py/pull/18/files#diff-f6de4b024452efaaaf2e7fbe1895b799R32-R43
I guess the whole part can be simplified into
```python
    # fast path if identical
    if img1 == img2:
        if output and not options["diff_mask"]:
            for i in range(image_len):
                draw_gray_pixel(img1, 4 * i, options["alpha"], output)
        return 0
```

Since you encountered this bug, can you provide example images so I can add them into testcase?

Reply:
I was a little reluctant to make this change since it adds the requirements that `img1` implements `object.__eq__(self, other)` and that this function can be called with `other=img2`.

However, I think those requirements are quite reasonable, so I went ahead and did the change like you suggested.

I also added a new test case which I am hoping can serve as a motivation to improve/control the "Anti-aliased Pixel and Intensity Slope Detector" algorithm.